packages/mui-material/src/Modal/Modal.test.js

```diff
@@ -880,4 +880,20 @@ describe('<Modal />', () => {
       );
     }).not.toErrorDev();
   });
+
+  it('should not override default onKeyDown', async () => {
+    const handleKeyDown = spy();
+    const handleClose = spy();
+
+    const { user } = render(
+      <Modal open onKeyDown={handleKeyDown} onClose={handleClose}>
+        <div tabIndex={-1} />
+      </Modal>,
+    );
+
+    await user.keyboard('{Escape}');
```

Comment:
How long does it take to run this test with this utils vs. firing a keydown event directly?

Reply:
Not sure if this is correct way.

`await user.keyboard('{Escape}')` took around 30ms (+/- 5ms)

![image](https://github.com/user-attachments/assets/12f08e7a-8fa6-483d-8de0-bcb75f0338bc)

   ```
    fireEvent.keyDown(getByTestId('modal'), {
      key: 'Escape',
    });
```

too around 4ms (+/- 1ms)


![image](https://github.com/user-attachments/assets/c4381831-fbe4-4a0f-93bf-a56488453f9c)

Reply:
Ok thanks, so about 20ms. If we scale this to out 10,000 tests: https://app.circleci.com/pipelines/github/mui/material-ui/139173/workflows/97427b1f-67e7-406f-aec4-1fdcd5fe84dc/jobs/750329. It could mean 200s, from 3 minutes to 6 minutes: https://app.circleci.com/pipelines/github/mui/material-ui/139173/workflows/97427b1f-67e7-406f-aec4-1fdcd5fe84dc/jobs/750329. I think to be really careful with this.

Reply:
I guess we should include clear guidelines on when to use one or the other in `test/README.md`. From discussions in past PRs it seemed that we were leaning towards using `user` more in new tests.

Reply:
We can potentially drastically speed up `userEvent` by setting it up with `delay: null`.

```tsx
userEvent.setup({ delay: null })
```

It looks like the semantics will change somewhat. i.e. it will yield to a macrotask instead of a microtask in between events. I did some testing and it seems to make up the bulk of the increase in time. The question will be whether this is a problem for us or not.

In the meantime I'm also opening https://github.com/testing-library/user-event/pull/1233